util/fetch/src/client.rs

```diff
@@ -199,16 +199,17 @@ impl Client {
 		})
 	}
 
-	fn background_thread(tx_start: TxStartup, rx_proto: mpsc::Receiver<ChanItem>) -> io::Result<thread::JoinHandle<()>> {
+	fn background_thread(tx_start: TxStartup, rx_proto: mpsc::Receiver<ChanItem>, is_light: bool) -> io::Result<thread::JoinHandle<()>> {
 		thread::Builder::new().name("fetch".into()).spawn(move || {
 			let mut core = match reactor::Core::new() {
 				Ok(c) => c,
 				Err(e) => return tx_start.send(Err(e)).unwrap_or(())
 			};
 
 			let handle = core.handle();
+			let number_dns_threads = if is_light { 1 } else { 4 };
```

Comment:
Instead of passing the flag down you could pass the number of threads as a parameter.

Reply:
That's better, will change it!